lib/logstash/inputs/beats.rb

```diff
@@ -156,6 +162,10 @@ def register
       configuration_error "Multiline codec with beats input is not supported. Please refer to the beats documentation for how to best manage multiline data. See https://www.elastic.co/guide/en/beats/filebeat/current/multiline-examples.html"
     end
 
+    # define ecs name mapping
+    @field_hostname = ecs_select[disabled: "host",                    v1: "[host][name]"]
+    @field_hostip   = ecs_select[disabled: "[@metadata][ip_address]", v1: "[@metadata][beat][source][ip]"]
```

Comment:
shouldn't this be rather at least a plural `[@metadata][beats][source][ip]` or `[input]` prefixed?

maybe it would be good to have a clear name-space convention used everywhere.
e.g. if there's input-http and filter-http then using `[@metadata][http][url][original]` might collide, thoughts?

Reply:
Reading at https://www.elastic.co/guide/en/ecs/current/ecs-host.html probably one place for `ip` and `name` are inside `host`, following the description "should be populated with details about the host on which the event happened," effectively the host where the event is happening is the host that's running the Beats process.

Reply:
that's a different discussion but a good point - LS being a bridge here for the Beats agent it does not make much sense to set `host.name`. maybe set it as a meta-data fiels for both cases in ECS mode as : 
- `[@metadata][input][beats][host][name]`
- and `[@metadata][input][beats][host][ip]`

... users could still use the field in the pipeline config and even copy the field if they need it but our advice would be to not fill `host` in ECS mode?

Reply:
From that perspective it sounds reasonable, however the `@metadata` are not forwarded to the downstream systems (like ES) but if the user want it could just rename it to something it consider usable in his use case, respecting the ECS.
Given this, also the idea to separate these metadata by `plugin_type` and `plugin_name`, like you suggested in the case of `[input][http]` and `[filter][http]` seems correct